eng/build.ps1

```diff
@@ -341,8 +341,13 @@ function GetCompilerTestAssembliesIncludePaths() {
   return $assemblies
 }
 
+function GetCompilerEndToEndTestAssemblyExclusion() {
+  $assemblies += " --exclude '^Microsoft\.CodeAnalysis\.CSharp\.EndToEnd\.UnitTests$'"
```

Comment:
Use of `+=` here makes it seem like we are modifying a global. But I think this is a local, so we should probably use `=` instead.